python/dgl/data/utils.py

```diff
@@ -350,3 +361,395 @@ def __len__(self):
+    It uses multi-hot encoding to encode these labels.
+
+    Parameters
+    ----------
+    category_inputs : list of list of str
+        input categorical features
+    norm: str, optional
+        Which kind of normalization is applied to the features.
+        Supported normalization ops include:
+
+        (1) None, do nothing.
+        (2) `col`, column-based normalization. Normalize the data
+        for each column:
+
+        .. math::
+            x_{ij} = \frac{x_{ij}}{\sum_{i=0}^N{x_{ij}}}
+
+        (3) `row`, row-based normalization. Normalize the data for
+        each row:
+
+        .. math::
+            x_{ij} = \frac{x_{ij}}{\sum_{j=0}^N{x_{ij}}}
+
+        Default: None
+
+    Note
+    ----
+    sklearn.preprocessing.MultiLabelBinarizer is used to convert
+    categorical features into a multilabel format.
+
+    Return
+    ------
+    numpy.array
+        The features in numpy array
+
+    Example
+    -------
+
+    >>> inputs = [['A', 'B', 'C',], ['A', 'B'], ['C'], ['A']]
+    >>> feats = parse_category_multi_feat(inputs)
+    >>> feats
+        array([[1.,1.,1.],[1.,1.,0.],[0.,0.,1.],[1.,0.,0.]])
+
+    """
+    mlb = MultiLabelBinarizer()
+    feat = mlb.fit_transform(category_inputs)
+
+    if norm == 'col':
+        return col_normalize(feat)
+    if norm == 'row':
+        return row_normalize(feat)
+    else:
+        return feat
+
+def parse_numerical_feat(numerical_inputs, norm=None):
+    """ Parse numerical features.
+
+    Parameters
+    ----------
+    numerical_inputs : list of float or list of list of float
+        input numerical features
+    norm: str, optional
+        Which kind of normalization is applied to the features.
+        Supported normalization ops include:
+
+        (1) None, do nothing.
+        (2) `col`, column-based normalization. Normalize the data
+        for each column:
+
+        .. math::
+            x_{ij} = \frac{x_{ij}}{\sum_{i=0}^N{|x_{ij}|}}
+
+        (3) `row`, sane as None
+
+    Return
+    ------
+    numpy.array
+        The features in numpy array
+
+    Example
+
+    >>> inputs = [[1., 0., 0.],[2., 1., 1.],[1., 2., -3.]]
+    >>> feat = parse_numerical_feat(inputs, norm='col')
+    >>> feat
+    array([[0.25, 0., 0.],[0.5, 0.33333333, 0.25],[0.25, 0.66666667, -0.75]])
+
+    """
+    feat = np.array(numerical_inputs, dtype='float')
+
+    if norm == 'col':
+        return float_col_l1_normalize(feat)
+    else:
+        return feat
+
+def parse_numerical_multihot_feat(input_feats, low, high, bucket_cnt, window_size, norm=None):
+    r""" Parse numerical features by matching them into
+        different buckets.
+
+    A bucket range based algorithm is used to convert numerical value into multi-hop
+    encoding features.
+
+    A numerical value range [low, high) is defined, and it is
+    divied into #bucket_cnt buckets. For a input V, we get its effected range as
+    [V - window_size/2, V + window_size/2] and check how many buckets it covers in
```

Comment:
Is there any reference of using this? To me make it one-hot makes more sense (make the interval open at one side)